### ISSUE TITLE
DataManagementSetup implementation should be independent

### ISSUE THREAD

**Issue by mpostol** [Drutol/CrossHMI#24]
From the description, I can guess that the project provides two independent implementations of the `DataRepository`, According to the ReferenceApplication example and the architecture presented in <https://commsvr.gitbook.io/ooi/reactive-communication/semanticdata#reactive-networking-application-architecture> both need independent implementation of the `DataManagementSetup`.  It promotes separation of concerns and reusability.  Unfortunately, I can find only one implementation of this class, namely `NetworkEventsManager`. 

It must be recognized as a data error, so some comments are required.

**Comment by Drutol** [comment]
That's true, I acknowledge that it should be separate, yet I have run into a bit of trouble which I have described here: https://github.com/mpostol/OPC-UA-OOI/issues/468

**Comment by mpostol** [comment]
Not sure if adding abstract may be recognized as a real relief. Anyway, it is done. I have also improved documentation in this respect.
